pkg/manager/impl/execution_manager_test.go

```diff
@@ -3847,6 +3922,43 @@ func TestGetExecutionConfigOverrides(t *testing.T) {
 		assert.Equal(t, launchPlanOutputLocationPrefix, execConfig.RawOutputDataConfig.OutputLocationPrefix)
 		assert.Equal(t, launchPlanLabels, execConfig.GetLabels().Values)
 	})
+	t.Run("request with empty security context and deprecated auth role and lp with security context", func(t *testing.T) {
+		request := &admin.ExecutionCreateRequest{
+			Project: workflowIdentifier.Project,
+			Domain:  workflowIdentifier.Domain,
+			Spec: &admin.ExecutionSpec{
+				SecurityContext: &core.SecurityContext{
+					RunAs: &core.Identity{
+						K8SServiceAccount: "",
+						IamRole:           "",
+					},
+				},
+				AuthRole: &admin.AuthRole{
+					AssumableIamRole:         deprecatedAuthRoleAssumableIamRole,
+					KubernetesServiceAccount: deprecatedAuthRoleK8sServiceAccount,
+				},
+			},
+		}
+		launchPlan := &admin.LaunchPlan{
+			Spec: &admin.LaunchPlanSpec{
+				Annotations:         &admin.Annotations{Values: launchPlanAnnotations},
+				Labels:              &admin.Labels{Values: launchPlanLabels},
+				RawOutputDataConfig: &admin.RawOutputDataConfig{OutputLocationPrefix: launchPlanOutputLocationPrefix},
+				SecurityContext: &core.SecurityContext{
+					RunAs: &core.Identity{
+						K8SServiceAccount: launchPlanK8sServiceAccount,
+					},
+				},
+				MaxParallelism: launchPlanMaxParallelism,
+			},
+		}
+		execConfig, err := executionManager.getExecutionConfig(context.TODO(), request, launchPlan)
+		assert.NoError(t, err)
+		assert.Equal(t, launchPlanMaxParallelism, execConfig.MaxParallelism)
+		assert.Equal(t, launchPlanK8sServiceAccount, execConfig.SecurityContext.RunAs.K8SServiceAccount)
```

Comment:
is this correct? if we set the deprecated field in the CreateExecutionRequest shouldn't that take precedence over whatever is in the launch plan?

Reply:
This gives security context the precedence if its sets on the launch plan.  

If its not set as in this test `request full config with deprecated auth role` then it uses the deprecated auth role.

Here the assumption is if user has migrated to launchplans with new way of passing security context then he should move to execution create requests also with the same or do we want to support this case and use the deprecated  fields from request

Reply:
hm I see what you're saying wrt the user has migrated on their end from entirely using auth role to security ctx, i just don't know if we can safely make that assumption. perhaps i'm being too defensive. we could document this somewhere user-facing perhaps